src/pages/home/sidebar/SubNavigation/SubNavigation.js

```diff
@@ -17,17 +26,117 @@ const propTypes = {
 };
 
 function SubNavigation({onLinkClick, insets}) {
+    const sidebarNavigation = useContext(SidebarNavigationContext);
+    const {translate} = useLocalize();
+
     useEffect(() => {
         Performance.markStart(CONST.TIMING.SIDEBAR_LOADED);
         Timing.start(CONST.TIMING.SIDEBAR_LOADED, true);
     }, []);
 
     return (
         <View style={styles.subNavigationContainer}>
-            <SidebarLinksData
-                insets={insets}
-                onLinkClick={onLinkClick}
-            />
+            <View style={[StyleSheet.absoluteFillObject, StyleUtils.displayIfTrue(sidebarNavigation.selectedSubNavigationMenu === CONST.SUB_NAVIGATION_MENU.CHATS)]}>
+                <SidebarLinksData
+                    insets={insets}
+                    onLinkClick={onLinkClick}
+                />
+            </View>
+            <View style={[StyleSheet.absoluteFillObject, StyleUtils.displayIfTrue(sidebarNavigation.selectedSubNavigationMenu === CONST.SUB_NAVIGATION_MENU.MONEY)]}>
+                <SubNavigationMenu
+                    title={translate('globalNavigation.money')}
+                    menuItems={[
+                        {
+                            icon: Expensicons.Receipt,
+                            text: translate('subNavigation.expenses'),
+                            value: SCREENS.EXPENSES_OLDDOT,
+                            onSelected: () => {
+                                Navigation.navigate(ROUTES.EXPENSES_OLDDOT);
+                            },
+                        },
+                        {
+                            icon: Expensicons.Document,
+                            text: translate('subNavigation.reports'),
+                            value: SCREENS.REPORTS_OLDDOT,
+                            onSelected: () => {
+                                Navigation.navigate(ROUTES.REPORTS_OLDDOT);
+                            },
+                        },
+                        {
+                            icon: Expensicons.Lightbulb,
+                            text: translate('subNavigation.insights'),
+                            value: SCREENS.INSIGHTS_OLDDOT,
+                            onSelected: () => {
+                                Navigation.navigate(ROUTES.INSIGHTS_OLDDOT);
+                            },
+                        },
+                    ]}
+                />
+            </View>
+            <View style={[StyleSheet.absoluteFillObject, StyleUtils.displayIfTrue(sidebarNavigation.selectedSubNavigationMenu === CONST.SUB_NAVIGATION_MENU.WORKSPACES)]}>
```

Comment:
Am I correct that we are returning all of these elements, but hiding them in the dom? Would it not be better to conditionally return them?

Reply:
Hiding them in the dom is faster than mounting and unmounting every time. That's especially true for the report list. Additionally the current implementation requires the report list to be mounted to exit the splash screen